flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/api/internal/ShowCreateUtilTest.java

```diff
@@ -0,0 +1,218 @@
+        final String createViewString =
+                ShowCreateUtil.buildShowCreateViewRow(resolvedCatalogView, VIEW_IDENTIFIER, false);
+        assertThat(createViewString).isEqualTo(expected);
+    }
+
+    private static Collection<Arguments> argsForShowCreateView() {
+        Collection<Arguments> argList = new ArrayList<>();
+        argList.add(
+                Arguments.of(
+                        createResolvedView(ONE_COLUMN_SCHEMA, "SELECT 1", "SELECT 1", null),
+                        "CREATE VIEW `catalogName`.`dbName`.`viewName` (\n"
+                                + "  `id`\n"
+                                + ") AS SELECT 1\n"));
+
+        argList.add(
+                Arguments.of(
+                        createResolvedView(
+                                TWO_COLUMNS_SCHEMA,
+                                "SELECT id, name FROM tbl_a",
+                                "SELECT id, name FROM `catalogName`.`dbName`.`tbl_a`",
+                                "View comment"),
+                        "CREATE VIEW `catalogName`.`dbName`.`viewName` (\n"
+                                + "  `id`,\n"
+                                + "  `name`\n"
+                                + ") COMMENT 'View comment'\n"
+                                + "AS SELECT id, name FROM `catalogName`.`dbName`.`tbl_a`\n"));
+        return argList;
+    }
+
+    private static Collection<Arguments> argsForShowCreateTable() {
+        Collection<Arguments> argList = new ArrayList<>();
+        argList.add(
+                Arguments.of(
+                        createResolvedTable(
+                                ONE_COLUMN_SCHEMA,
+                                Collections.emptyMap(),
+                                Collections.emptyList(),
+                                TableDistribution.of(
+                                        TableDistribution.Kind.HASH,
+                                        2,
+                                        Arrays.asList("key1", "key2")),
+                                null),
+                        "CREATE TABLE `catalogName`.`dbName`.`tableName` (\n"
+                                + "  `id` INT\n"
+                                + ") DISTRIBUTED BY HASH(`key1`, `key2`) INTO 2 BUCKETS\n"));
+
+        argList.add(
+                Arguments.of(
+                        createResolvedTable(
+                                ONE_COLUMN_SCHEMA,
+                                Collections.emptyMap(),
+                                Collections.emptyList(),
+                                TableDistribution.of(
+                                        TableDistribution.Kind.RANGE, 2, Arrays.asList("1", "10")),
+                                "Table comment"),
+                        "CREATE TABLE `catalogName`.`dbName`.`tableName` (\n"
+                                + "  `id` INT\n"
+                                + ") COMMENT 'Table comment'\n"
```

Comment:
It's weird that COMMENT causes a newline at the end whereas other subclauses don't. We should fix this or we will never fix it. Can we add a `\n` after `)`:

```
CREATE TABLE `catalog`.`database`.`t` (
  `s` VARCHAR(2147483647)
)
DISTRIBUTED INTO 6 BUCKETS
WITH (
  'k' = 'v'
)
```

```
CREATE TABLE `catalog`.`database`.`t` (
  `s` VARCHAR(2147483647)
)
COMMENT 'comment'
DISTRIBUTED INTO 6 BUCKETS
WITH (
  'k' = 'v'
)
```

```
CREATE TABLE `catalog`.`database`.`t` (
  `s` VARCHAR(2147483647)
)
COMMENT 'comment'
DISTRIBUTED INTO 6 BUCKETS
PARTITIONED BY (`key1`, `key2`)
WITH (
  'k' = 'v'
)
```

```
CREATE VIEW `catalogName`.`dbName`.`viewName` (
  `id`
  `name`
)
COMMENT 'View comment'
AS SELECT ...
```